src/xenia/kernel/kernel_state.cc

```diff
@@ -316,6 +317,24 @@ void KernelState::SetExecutableModule(object_ref<UserModule> module) {
     *variable_ptr = executable_module_->hmodule_ptr();
   }
 
+  // Setup the kernel's ExLoadedImageName field
+  export_entry = processor()->export_resolver()->GetExportByOrdinal(
+      "xboxkrnl.exe", ordinals::ExLoadedImageName);
+
+  if (export_entry) {
+    char* variable_ptr =
+        memory()->TranslateVirtual<char*>(export_entry->variable_ptr);
+    size_t length = 0;
+
+    if (module) {
+      std::string image_name(module->path());
+      length = std::min(size_t(XFile::kMaximalFileNameLength - 1),
+                        image_name.size());
+      std::memcpy(variable_ptr, image_name.c_str(), length);
+    }
+    variable_ptr[length] = '\0';
```

Comment:
We have xe::string_util::copy_truncating now for this :)

Reply:
Changed ;) Also figured out a little mistake I've made last time 🐑